### PR TITLE
feat: add login as guest

### PR DESCRIPTION
**Pull request by TMNhat1810**
contribute for:  [US-3]

### CODE REVIEW

File: app/services/passport.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+passport.use(
+    new GoogleStrategy(
+        {
+            clientID: CLIENT_ID,
+            clientSecret: CLIENT_SECRET,
+            callbackURL: `${SERVER_URL}/auth/google/callback`,
+            passReqToCallback: true,
+        },
+        async (request, accessToken, refreshToken, profile, done) => {
+            let data = profile._json;
```
Comment (ces-anduynguyen) [inline app/services/passport.js:18]: should be constant

File: app/controllers/auth.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        let { username } = req.body;
+
+        if (!username) {
+            return res.status(HTTP_STATUS.BAD_REQUEST.BAD).json({
+                success: false,
+            });
+        }
+
+        try {
+            let user = new User({
```
Comment (ces-anduynguyen) [inline app/controllers/auth.js:18]: should be constant

File: app/controllers/auth.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { User } from '../models/index.js';
+import HTTP_STATUS from '../../constants/HTTPStatusCode.js';
+import { USER_TYPES } from '../../constants/db.constants.js';
+import MESSAGE from '../../constants/message.js';
+
+export default {
```
Comment (ces-anduynguyen) [inline app/controllers/auth.js:6]: why default? should be class or const instead.
Read more [here](https://basarat.gitbook.io/typescript/main-1/defaultisbad#poor-discoverability)

File: app/services/passport.js (hunk 1 above)
Comment (TMNhat1810) [reply app/services/passport.js]: this code belong to google-login, I have splited it. please check!

File: app/models/room.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import mongoose from 'mongoose';
```
[the remark this answers is not in the document]
Comment (TMNhat1810) [reply app/models/room.js]: this code belong to init-database, I have splited it. please check!

File: app/controllers/auth.js (hunk 2 above)
Comment (TMNhat1810) [reply app/controllers/auth.js]: i did it!

File: app/controllers/auth.js (hunk 3 above)
Comment (TMNhat1810) [reply app/controllers/auth.js]: i did it!

File: app/controllers/auth.js [hunk 5]
```diff
@@ -33,4 +35,27 @@ export const authController = {
...
+      const user = await UserService.createGuestUser(username);
+
+      return res.status(HTTP_STATUS.OK).json({
+        success: true,
+        data: user,
+      });
+    } catch (err) {
+      return res.status(HTTP_STATUS.INTERNAL_SERVER_ERROR).json({
+        success: false,
+        message: RESPONSE_MESSAGE.INTERNAL_SERVER_ERROR,
```
Comment (ces-hoangdao) [inline app/controllers/auth.js:56]: I ask Long to define an util func let talk with him and use it

**Review by ces-hoangdao — APPROVED** [review]

**Pull request merged**